HeartbeatLogging/Sources/HeartbeatController.swift

```diff
@@ -14,92 +14,104 @@
+  // TODO: Decide on default value.
+  /// The max capacity of heartbeats to store in storage.
+  private let heartbeatsStorageCapacity: Int = 30
+  /// Current date provider. It is used for testability.
   private let dateProvider: () -> Date
-  // TODO: Verify that this standardization aligns with backend.
+  // TODO: Maybe share config with HeartbeatsPayload's DateFormatter?
   /// Used for standardizing dates for calendar-day comparision.
-  static let dateStandardizer = Calendar(identifier: .gregorian).startOfDay(for:)
+  static let dateStandardizer: (Date) -> (Date) = {
+    var calendar = Calendar(identifier: .iso8601)
+    calendar.locale = Locale(identifier: "en_US_POSIX")
+    calendar.timeZone = TimeZone(secondsFromGMT: 0)!
+    return calendar.startOfDay(for:)
+  }()
 
   /// Public initializer.
-  ///
-  /// - Parameter id: The `id` to associate this logger's internal storage with.
+  /// - Parameter id: The `id` to associate this controller's heartbeat storage with.
   public convenience init(id: String) {
-    // TODO: Sanitize id.
+    self.init(id: id, dateProvider: Date.init)
+  }
+
+  /// Convenience initializer. Mirrors the semantics of the public intializer with the added benefit of
+  /// injecting a custom date provider for improved testability.
+  /// - Parameters:
+  ///   - id: The id to associate this controller's heartbeat storage with.
+  ///   - dateProvider: A date provider.
+  convenience init(id: String,
+                   dateProvider: @escaping () -> Date) {
     let storage = HeartbeatStorage.getInstance(id: id)
-    self.init(storage: storage)
+    self.init(storage: storage, dateProvider: dateProvider)
   }
 
   /// Designated initializer.
-  ///
   /// - Parameters:
-  ///   - storage: The logger's internal storage object.
-  ///   - dateProvider: TODO: Document.
+  ///   - storage: A heartbeat storage container.
+  ///   - dateProvider: A date provider. Defaults to providing the current date.
   init(storage: HeartbeatStorageProtocol,
        dateProvider: @escaping () -> Date = Date.init) {
     self.storage = storage
     self.dateProvider = { Self.dateStandardizer(dateProvider()) }
   }
 
-  /// Asynchronously log a new heartbeat, if needed.
+  /// Asynchronously logs a new heartbeat, if needed.
   ///
   /// - Note: This API is thread-safe.
   ///
-  /// - Parameter agent: A `String` identifier to associate a new heartbeat with.
+  /// - Parameter agent: The string agent to associate the logged heartbeat with.
   public func log(_ agent: String) {
     let date = dateProvider()
-    let capacity = limit
+    let capacity = heartbeatsStorageCapacity
 
     storage.readAndWriteAsync { heartbeatInfo in
       var heartbeatInfo = heartbeatInfo ?? HeartbeatInfo(capacity: capacity)
 
+      // Filter for the time periods where the last heartbeat to be logged for
+      // that time period was logged more than one time period (i.e. day) ago.
```

Comment:
The `cache` probably needs more descriptive name. I like that comment structure. I will work on it.

Reply:
How does ⬇️  sound?
> The cache contains the last added date for each time period. We only need to cache the date since it's the only piece of information used to determine whether or not to append a new heartbeat. 

The cache can be used by clients (as done here) to decide what time periods need a new heartbeat. 

WDYT is most helpful? 
- renaming `cache` to `lastAddedHeartbeatDateByTimePeriod` (I like the conciseness of `cache`, esp. in the `HeartbeatInfo` implementation.)
- Adding additional documentation here
- Adding additional documentation to the `cache` property declaration in `HeartbeatInfo`

Reply:
I added the comment to the `cache` property declaration. I think the hover doc provides a nice glimpse into what it is/does:

<img width="450" alt="Screen Shot 2021-12-08 at 5 26 42 PM" src="https://user-images.githubusercontent.com/36927374/145295617-7b06384d-3723-446c-ae84-e57e0290d305.png">

Reply:
Thanks for the naming suggestions! Heh, I went back and forth between a the concise `cache` vs a more self-documenting name. PTAL at commit abbaa4f18 and lmk what you think.